src/Base/Vector.cpp

```diff
@@ -0,0 +1,90 @@
+/* Copyright 2022 The AMReX Community
+ *
+ * Authors: Ryan Sandberg
+ * License: BSD-3-Clause-LBNL
+ */
+#include <pybind11/pybind11.h>
+#include <pybind11/operators.h>
+#include <pybind11/numpy.h>
+#include <pybind11/stl.h>
+
+#include <AMReX_Config.H>
+#include <AMReX_PODVector.H>
+
+#include <sstream>
+#include <string>
+#include <type_traits>
+#include <optional>
+#include <vector>
+
+namespace py = pybind11;
+using namespace amrex;
+
+
+template <class T, class Allocator = std::allocator<T> >
+void make_Vector(py::module &m, std::string typestr)
+{
+    using Vector_type=Vector<T, Allocator>;
+    auto const v_name = std::string("Vector_").append(typestr);
+    
+    py::class_<Vector_type>(m, v_name.c_str())
+        .def("__repr__",
+             [typestr](Vector_type const & v) {
+                 std::stringstream s, rs;
+                 s << v.size();
+                 rs << "<amrex.Vector of type '" + typestr +
+                        "' and size '" + s.str() + "'>\n";
+                 rs << "[ ";
+                 for (int ii = 0; ii < int(v.size()); ii++) {
+                     rs << v[ii] << " ";
+                 }
+                 rs << "]\n";
+                 return rs.str();
+             }
+        )
+        .def(py::init<>())
+
+        /* init from a numpy or other buffer protocol array: non-owning view
```

Comment:
I don't think that an AMReX or std vector can support a non-owning view on other data, thus please remove the block here.

Reply:
done